pkgs/development/libraries/science/math/libamplsolver/default.nix

```diff
@@ -9,19 +9,23 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-LVmScuIvxmZzywPSBl9T9YcUBJP7UFAa3eWs9r4q3JM=";
   };
 
-  patches = [
+  patchesLinux = [
     # Debian provides a patch to build a shared library
     (fetchpatch {
       url = "https://sources.debian.org/data/main/liba/libamplsolver/0~20190702-2/debian/patches/fix-makefile-shared-lib.patch";
       sha256 = "sha256-96qwj3fLugzbsfxguKMce13cUo7XGC4VUE7xKcJs42Y=";
     })
   ];
 
+  patchesDarwin = [ ./libamplsolver-darwin.patch ];
```

Comment:
Where did this come from?

Reply:
I wrote it based on the Debian patch. Incantation to generate the dylib is from [here](https://stackoverflow.com/questions/25321911/convert-a-to-dylib-in-mac-osx).